app/src/main/java/fr/free/nrw/commons/CommonsApplication.java

```diff
@@ -339,4 +345,47 @@ public interface LogoutListener {
 
         void onLogoutComplete();
     }
+
+    public static class BaseLogoutListener implements CommonsApplication.LogoutListener {
```

Comment:
It occurs to me that we coukd actually refactor the listeners into two for better code readability. Here's a commit to give you an idea of what I mean: https://github.com/sivaraam/apps-android-commons/commit/816d932897dfb6eeceef69e9c05fb87645d0430e

Do feel free to integrate it or share your thoughts on the idea.